src/test/java/com/openlattice/rehearsal/chronicle/ChronicleTestBase.kt

```diff
@@ -0,0 +1,213 @@
+                .of("com.google.android.gm", "Gmail")
+        val YOUTUBE = Pair
+                .of("com.google.android.youtube", "YouTube")
+        val CHROME = Pair
+                .of("com.android.chrome", "Chrome")
+        val MAPS = Pair
+                .of("com.google.android.apps.maps", "Maps")
+        val PHONE = Pair
+                .of("com.android.dialer", "Phone")
+
+        // for data integration permissions
+        private val ORGANIZATION_ID = UUID.fromString("00000000-0000-0001-0000-000000000000")
+        private val USERS_ROLE = "OpenLattice User Role"
+        private val CHRONICLE_NAME = "chronicle_"
+
+
+        var entitySetNameIdMap: Map<String, UUID> = HashMap()
+        lateinit var fullNamePTID: UUID
+        lateinit var durationPTID: UUID
+        lateinit var titlePTID: UUID
+        lateinit var startDateTimePTID: UUID
+        lateinit var dateLoggedPTID: UUID
+        lateinit var recordTypePTID: UUID
+        lateinit var subjectIdPTID: UUID
+        lateinit var statusPTID: UUID
+        lateinit var olIdPTID: UUID
+        lateinit var participantEntityType: EntityType
+        lateinit var auditEntityTypeId: UUID
+
+        lateinit var participant1EntityKeyId: UUID
+        lateinit var participant2EntityKeyId: UUID
+        lateinit var participant3EntityKeyId: UUID
+
+        lateinit var device1EntityKeyId: UUID
+        lateinit var device2EntityKeyId: UUID
+        lateinit var device3EntityKeyId: UUID
+
+        @JvmStatic
+        @BeforeClass
+        fun init() {
+            loginAs("admin")
+            getPropertyTypeIdsAndEntitySets()
+            getDataVariables()
+            deleteEntities()
+        }
+
+        fun deleteEntities() {
+            val entitySetsIds = HashSet<UUID>()
+            entitySetsIds.add(entitySetNameIdMap[CHRONICLE_USER_APPS]!!)
+            for (entitySetId in entitySetsIds) {
+                dataApi!!.deleteAllEntitiesFromEntitySet(entitySetId, DeleteType.Hard)
+            }
+        }
+
+        fun getEntityKeyIds(entitySetId: UUID, key: UUID, entityKey: String): UUID {
+            return dataIntegrationApi.getEntityKeyIds(setOf(
+                    EntityKey(
+                            entitySetId,
+                            ApiUtil.generateDefaultEntityId(mutableListOf(key), mapOf(key to setOf(entityKey)))
+                    ))).first()
+        }
+
+        private fun getDataVariables() {
+            val participantEntitySetId = entitySetsApi.getEntitySetId(PARTICIPANTS_PREFIX + STUDY_ID)
+
+            participant1EntityKeyId = getEntityKeyIds(participantEntitySetId, subjectIdPTID!!, PARTICIPANT1)
+            participant2EntityKeyId = getEntityKeyIds(participantEntitySetId, subjectIdPTID!!, PARTICIPANT2)
+            participant3EntityKeyId = getEntityKeyIds(participantEntitySetId, subjectIdPTID!!, PARTICIPANT3)
+
+            device1EntityKeyId = getEntityKeyIds(entitySetNameIdMap[DEVICES_ENTITY_SET_NAME]!!, olIdPTID!!, DEVICE1)
```

Comment:
should be able to remove the `!!` after `olIdPTID` and if you use `entitySetNameIdMap.getValue()` instead of `entitySetNameIdMap[]` then you should be able to remove all of the `!!` here